from fastapi import HTTPException, status

class BusyLogin(HTTPException):
    def __init__(self):
        super().__init__(status_code=status.HTTP_400_BAD_REQUEST, detail="Это имя пользователя уже занято")

class WrongLogin(HTTPException):
    def __init__(self):
        super().__init__(status_code=status.HTTP_400_BAD_REQUEST, detail="Пользователь с таким логином не зарегистрирован")

class WrongPassword(HTTPException):
    def __init__(self):
        super().__init__(status_code=status.HTTP_400_BAD_REQUEST, detail="Неправильный пароль")

class WrongToken(HTTPException):
    def __init__(self):
        super().__init__(status_code=status.HTTP_400_BAD_REQUEST, detail="Неправильный токен")
        
class DogDoesntExist(HTTPException):
    def __init__(self):
        super().__init__(status_code=status.HTTP_400_BAD_REQUEST, detail="Такой собаки нет в базе данных")

class UserDoesntExist(HTTPException):
    def __init__(self):
        super().__init__(status_code=status.HTTP_400_BAD_REQUEST, detail="Такого пользователя нет в базе данных")

class ExistSubscr(HTTPException):
    def __init__(self):
        super().__init__(status_code=status.HTTP_400_BAD_REQUEST, detail="Вы уже подписаны на эту собаку")

class SubscrDoesntExist(HTTPException):
    def __init__(self):
        super().__init__(status_code=status.HTTP_400_BAD_REQUEST, detail="Вы не были подписаны на эту собаку")
